### PR TITLE
docs: add docs for workspace variants

### PR DESCRIPTION
**Pull request by debs-obrien**
@giladshoham @davidfirst these have been copied from docs and will need reviewing before merging to make sure everything is correct

## Proposed Changes

-
-
-

### CODE REVIEW

File: scopes/workspace/variants/variants.docs.md
Comment (GiladShoham) [inline scopes/workspace/variants/variants.docs.md:80]: The propagate false in this case on `components/react` will means that components under 
`components/react/ui` will have config from both rules `components/react/ui` and `components/react` but not from `components`. so the place where the propagate false is defined are also included. so if you don't want to get config from `components/react` the propagate false should be defined in the `components/react/ui`

File: scopes/workspace/variants/variants.docs.md
Comment (GiladShoham) [inline scopes/workspace/variants/variants.docs.md:7]: one of the most important things to mention here is how this merge between different rules is working:
we took aspects from each level and add them, but once we see an aspect we won't merge it's configuration if it's defined in an upper level again with different config (let's talk about the rational of it, because many developers already asks me why it's like this)
so for example:

`workspace.jsonc`
```json
{
  "teambit.workspace/variants": {
    "*": {
        "my-aspect1": {
          "aspect1-root-key": "aspect1-root-val"
        },
       "my-aspect2": {
        "aspect2-root-key": "aspect2-root-val"
      }
        "my-aspect4": {
          "aspect4-root-key": "aspect4-root-val"
        }
    },
    "components": {
      "my-aspect1": {
        "aspect1-components-key": "aspect1-components-val"
      },
      "my-aspect2": {
        "aspect2-components-key": "aspect2-components-val"
      }
    },
    "components/ui": {
      "my-aspect1": {
        "aspect1-components-ui-key": "aspect1-components-ui-val"
      },
      "my-aspect3": {
        "aspect3-components-ui-key": "aspect3-components-ui-val"
      }
    }
  }
}
```
`components/ui/button final config`
```json
{
  "my-aspect1": {
    "aspect1-components-ui-key": "aspect1-components-ui-val"
  },
  "my-aspect2": {
    "aspect2-components-key": "aspect2-components-val"
  },
  "my-aspect3": {
    "aspect3-components-ui-key": "aspect3-components-ui-val"
  },
  "my-aspect4": {
    "aspect4-root-key": "aspect4-root-val"
  }
}
```

As you can see, we didn't took the `aspect1-components-ui-key` nor the `aspect1-root-key` (because we already saw `my-aspect1` in a more specific rule.
We also didn't took the `aspect2-root-key` because we already saw it in the `components` rule

**Review by GiladShoham — CHANGES_REQUESTED** [review]

File: scopes/workspace/variants/variants.docs.md
Comment (GiladShoham) [inline scopes/workspace/variants/variants.docs.md:9]: maybe it's also worth mentioning the relation between compoennts.json and variants

**Review by GiladShoham — CHANGES_REQUESTED** [review]

**Review by GiladShoham — APPROVED** [review]

**Review by itaymendel — APPROVED** [review]

**Pull request merged**
